crates/task-impls/src/quorum_proposal_recv.rs

```diff
@@ -0,0 +1,164 @@
+
+    /// timeout task handle
+    pub timeout_task: Option<JoinHandle<()>>,
+
+    /// View timeout from config.
+    pub timeout: u64,
+
+    /// Output events to application
+    pub output_event_stream: async_broadcast::Sender<Event<TYPES>>,
+
+    /// This node's storage ref
+    pub storage: Arc<RwLock<I::Storage>>,
+
+    /// The most recent proposal we have, will correspond to the current view if Some()
+    /// Will be none if the view advanced through timeout/view_sync
+    pub current_proposal: Option<QuorumProposal<TYPES>>,
+
+    /// Spawned tasks related to a specific view, so we can cancel them when
+    /// they are stale
+    pub spawned_tasks: BTreeMap<TYPES::Time, Vec<JoinHandle<()>>>,
+
+    /// The node's id
+    pub id: u64,
+}
+
+#[cfg(feature = "dependency-tasks")]
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumProposalRecvTaskState<TYPES, I> {
+    /// Cancel all tasks that have been spawned before the provided view.
+    async fn cancel_tasks(&mut self, view: TYPES::Time) {
+        let keep = self.spawned_tasks.split_off(&view);
+        let mut cancel = Vec::new();
+        while let Some((_, tasks)) = self.spawned_tasks.pop_first() {
+            let mut to_cancel = tasks.into_iter().map(cancel_task).collect();
+            cancel.append(&mut to_cancel);
+        }
+        self.spawned_tasks = keep;
+        join_all(cancel).await;
+    }
+
+    /// Handles all consensus events relating to propose and vote-enabling events.
+    #[instrument(skip_all, fields(id = self.id, view = *self.cur_view), name = "Consensus replica task", level = "error")]
+    pub async fn handle(
+        &mut self,
+        event: Arc<HotShotEvent<TYPES>>,
+        event_stream: Sender<Arc<HotShotEvent<TYPES>>>,
+    ) {
+        match event.as_ref() {
+            HotShotEvent::QuorumProposalRecv(proposal, sender) => {
+                match handle_quorum_proposal_recv(proposal, sender, event_stream.clone(), self)
+                    .await
+                {
+                    Ok(Some(current_proposal)) => {
+                        self.current_proposal = Some(current_proposal);
+
+                        // Build the parent leaf since we didn't find it during the proposal check.
+                        let consensus = self.consensus.read().await;
+
+                        broadcast_event(
+                            Arc::new(HotShotEvent::VoteNow(
+                                proposal.data.get_view_number() + 1,
+                                VoteDependencyData {
+                                    quorum_proposal: current_proposal,
+                                },
+                            )),
+                            &event_stream,
+                        )
+                        .await;
+                        // if self.vote_if_able(&event_stream).await {
+                        //     self.current_proposal = None;
+                        // }
```

Comment:
I think we should still reset `current_proposal` though we don't have to call `vote_if_able` here.

Reply:
Actually, do we need `self.curren_proposal`? Looks like we don't use it anywhere in this task.

Reply:
Yeah we can blow it away, it was legacy for when I was trying to make this work within the Consensus task.